### PR TITLE
Use `fullname` from `$admin` user info instead of deprecated `firstname`/`lastname` arguments

### PR DESCRIPTION
**Pull request by daniel-fahey**
## Problem

Emails from the app are sent with "null null" as the sender display name
- `MAIL_FROM_NAME="null null"` in `.env`
- Manual edits to `.env` are overwritten during upgrade

## Reason

`$admin` user's info `firstname`/`lastname` keys are deprecated and no longer available [since YunoHost version 11.1](https://github.com/YunoHost/yunohost/pull/1516), so when they are retrieved by
```
email_firstname="$(ynh_user_get_info --username=$admin --key=firstname)"
email_lastname="$(ynh_user_get_info --username=$admin --key=lastname)"
```
and set by
```
ynh_app_setting_set --app=$app --key=email_firstname --value="$email_firstname"
ynh_app_setting_set --app=$app --key=email_lastname --value="$email_lastname"
```
`email_firstname` and `email_lastname` app settings are set to `null`.

- During install and upgrade `ynh_add_config` replaces `MAIL_FROM_NAME="__EMAIL_FIRSTNAME__ __EMAIL_LASTNAME__ "` with  `MAIL_FROM_NAME="null null"`

## Solution

- Use `fullname` keys instead of `firstname`/`lastname`
- Update app settings during upgrade to handle the deprecation
  - Feedback on this very welcome

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by rndmh3ro** [comment]
!testme

File: scripts/upgrade [hunk 1]
```diff
@@ -45,6 +45,25 @@ ynh_add_config --template="cron" --destination="/etc/cron.d/$app"
...
+	ynh_app_setting_delete --app=$app --key=email_lastname
+
+	# Retrieve admin user settings
+	email_fullname="$(ynh_user_get_info --username=$admin --key=fullname)"
+	email="$(ynh_user_get_info --username=$admin --key=mail)" # include as admin's email address is also subject to change
+
+	# Set
+	ynh_app_setting_set --app=$app --key=email_fullname --value="$email_fullname"
+	ynh_app_setting_set --app=$app --key=email --value=$email
+fi
```
Comment (ericgaspar) [inline scripts/upgrade:65]:
```suggestion
#=================================================
# ENSURE DOWNWARD COMPATIBILITY
#=================================================
ynh_script_progression --message="Ensuring downward compatibility..." --weight=1

ynh_app_setting_delete --app=$app --key=email_firstname
ynh_app_setting_delete --app=$app --key=email_lastname
	
email_fullname="$(ynh_user_get_info --username=$admin --key=fullname)"
email="$(ynh_user_get_info --username=$admin --key=mail)" 
```

Comment (daniel-fahey) [reply scripts/upgrade]: Thanks @ericgaspar, in 3e4c879655c6b1c2d93893915ef7951bfacc4f60 I've renamed the comment section, added in the progression info, and removed the conditional on `$upgrade_type`, but I really think it's a good idea keep the upgraded app's state in-line with a fresh install to help future development and simplify troubleshooting for users.

File: scripts/install [hunk 2]
```diff
@@ -28,8 +27,7 @@ ynh_script_progression --message="Storing installation settings..."
 ynh_app_setting_set --app=$app --key=api_secret --value=$api_secret
 ynh_app_setting_set --app=$app --key=app_key --value=$app_key
 ynh_app_setting_set --app=$app --key=phantomjs_key --value=$phantomjs_key
-ynh_app_setting_set --app=$app --key=email_firstname --value="$email_firstname"
-ynh_app_setting_set --app=$app --key=email_lastname --value="$email_lastname"
+ynh_app_setting_set --app=$app --key=email_fullname --value="$email_fullname"
 ynh_app_setting_set --app=$app --key=email --value=$email
```
Comment (ericgaspar) [inline scripts/install:31]:
```suggestion
```

File: scripts/upgrade (hunk 1 above)
Comment (ericgaspar) [reply scripts/upgrade]: We don't want to store those.
```
ynh_app_setting_set --app=$app --key=email_fullname --value="$email_fullname"
ynh_app_setting_set --app=$app --key=email --value=$email
```
We just call them when needed

**Comment by daniel-fahey** [comment]
> The `.env` file may need to regenerated for `change_url` script so that `.env` gets updated with those modifications.

Thanks @ericgaspar, yeah the `change_url` script [currently](https://github.com/YunoHost-Apps/invoiceninja5_ynh/blob/1d9c7314d73c42474dbbff70863eb943559e7b36/scripts/change_url#L26) doesn't get or set the app's settings state (i.e. `/etc/yunohost/apps/invoiceninja5/settings.yml`) for `$email`, `$email_fullname`, `$domain` or otherwise. It just uses `ynh_replace_string()` directly to replace any instance of `$old_domain$old_path`, and as an unwanted side-effect it may well change email addresses that also match when `$old_path` is empty. It'd be better to let `ynh_add_config()` use `$domain` (if it's available), but maybe that's for a different pull request, I think.

For this PR I think we could stop using the app's state to store `$email` nor `$email_fullname` and get them from the live system as needed. Please check out my new commit 973cc41c1de98446067ca35e03ee456caf028982; fewer moving parts, less state to manage.

**Comment by rndmh3ro** [comment]
Sorry!

**Comment by daniel-fahey** [comment]
No worries! Anything I can do to help get this merged?

**Comment by rndmh3ro** [comment]
!testme

**Comment by daniel-fahey** [comment]
Thanks @rndmh3ro, also note the `__PATH_URL__` doesn't work either. According to [Packaging v2](https://yunohost.org/en/packaging_v2) I think it should just be `__PATH__`. For another time in another PR anyway.

**Pull request merged**